Gems/ROS2/Code/Source/EffortSensor/ROS2EffortSensorComponent.cpp

```diff
@@ -0,0 +1,108 @@
+namespace ROS2
+{
+    namespace Internal
+    {
+        const char* kWrenchMsgType = "geometry_msgs::msg::WrenchStamped";
+    }
+
+    void ROS2EffortSensorComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ROS2EffortSensorComponent, ROS2SensorComponent>()->Version(1);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ROS2EffortSensorComponent>("ROS2 Effort Sensor", "Effort sensor component")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"));
+            }
+        }
+    }
+
+    AZ::Entity::ComponentArrayType ROS2EffortSensorComponent::findCompatibleJointComponents(){
+        AZ::Entity::ComponentArrayType components;
+        for (auto type : CompatibleJointComponents ){
+            auto foundComponents = m_entity->FindComponents(type);
+            AZStd::move(foundComponents.begin(), foundComponents.end(), AZStd::back_inserter(components));
+        }
+        return components;
+    }
+
+    void ROS2EffortSensorComponent::Activate()
+    {
+        AZ::Entity::ComponentArrayType componentPtrVec = findCompatibleJointComponents();
+        AZ_Warning("ROS2EffortSensorComponent", componentPtrVec.size()>0, "ROS2EffortSensorComponent can work with only one joint component.");
+        AZ_Warning("ROS2EffortSensorComponent", componentPtrVec.size()==0, "ROS2EffortSensorComponent needs compatible joint");
+        if (componentPtrVec.size()>0){
+            m_jointId = AZ::EntityComponentIdPair(m_entity->GetId(), componentPtrVec.front()->GetId());
+        }
+        ROS2SensorComponent::Activate();
+
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+        AZ_Assert(m_sensorConfiguration.m_publishersConfigurations.size() == 1, "Invalid configuration of publishers for Effort sensor");
+
+        const auto publisherConfig = m_sensorConfiguration.m_publishersConfigurations[Internal::kWrenchMsgType];
+        const auto fullTopic = ROS2Names::GetNamespacedName(GetNamespace(), publisherConfig.m_topic);
+        m_effortPublisher = ros2Node->create_publisher<geometry_msgs::msg::WrenchStamped>(fullTopic.data(), publisherConfig.GetQoS());
+    }
+    ROS2EffortSensorComponent::ROS2EffortSensorComponent()
+    {
+        TopicConfiguration tc;
+        const AZStd::string type = Internal::kWrenchMsgType;
+        tc.m_type = type;
+        tc.m_topic = "effort";
+        m_sensorConfiguration.m_frequency = 10;
+        m_sensorConfiguration.m_publishersConfigurations.insert(AZStd::make_pair(type, tc));
+    }
+
+    void ROS2EffortSensorComponent::FrequencyTick()
+    {
+        AZStd::pair<AZ::Vector3, AZ::Vector3> wrench{0,0};
```

Comment:
nit: Not sure how/if this will compile on all platforms, we may want to use `AZ::Vector3::CreateZero()` instead?